server/api/google_forms_handler.post.ts

```diff
@@ -0,0 +1,122 @@
+}
+//the robot numer must be unique
+function insertUpdateRobot(robot: Robot, team_id: number) {
+  return sql<RobotTable>`robocomp.fn_insert_update_robot(robocomp.fn_get_next_robot_no(),${sql.lit(robot.name)},2024::smallint,${sql.lit(team_id)},${sql.lit(robot.competition)})`
+}
+
+function connectTeamParticipant(participant_id: number, team_id: number, role: string) {
+  return sql<TeamsParticipantsTable>`robocomp.fn_connect_participant_and_team(${sql.lit(participant_id)},${sql.lit(team_id)},${sql.lit(role)})`
+}
+
+function getNextRobotNo() {
+  return sql`robocomp.fn_get_next_robot_no()`
+}
+
+export default defineEventHandler(async (event) => {
+  const rawBody = await readBody(event)
+  if (!isFormsResponse(rawBody)) {
+    throw 'Invalid form data'
+  }
+  const body = rawBody as FormsResponse
+
+  const db = createKysely<Database>()
+
+  // there are no clear ways to check if google forms was edited, so this will have to be done by checking if the participant email was already used.
+  try {
+    const participant_team_id = await db
+      .selectFrom('robocomp.participants')
+      .select('id')
+      .where('email', '=', body.participants[0].email)
+      .execute()
+    if (participant_team_id.length !== 0) {
+      throw 'Edits are not yet supported'
+    }
+    // Create a team
+    const team = await sql<TeamRow[]>`SELECT * FROM ${insertUpdateTeam(body.team_name)}`.execute(db)
+    if (!team || team.rows.length === 0) {
+      throw 'Something went wrong while creating team'
+    }
+    //@ts-ignore, Idk why but it returns a different structure
+    const team_id = team.rows[0].fn_insert_update_team
+    const emails = body.participants.map((p) => p.email)
+    // Check if participants are already registered
+    const existing_participants = await db
+      .selectFrom('robocomp.teams_participants as tp')
+      .innerJoin('robocomp.teams as t', 'tp.teams_id', 't.id')
+      .innerJoin('robocomp.participants as p', 'tp.participants_id', 'p.id')
+      .select('p.id')
+      .where('t.year', '=', new Date().getFullYear())
+      .where('p.email', 'in', emails)
+      .execute()
+    if (existing_participants.length !== 0) {
+      throw 'Some of the participants are already registered'
+    }
+    // add participants
+    // I don't know how (if that even is possible to insert all participants as a single query with postgresql function)
+    let participants = []
+    try {
+      const participants_response = await Promise.all(
+        body.participants.map((p) => sql<ParticipantRow>`SELECT * FROM ${insertUpdateParticipant(p)}`.execute(db))
+      )
+      // @ts-ignore
+      participants = participants_response.map((pr) => pr.rows[0].fn_insert_update_participant)
```

Comment:
change from `sql<ParticipantRow[]>` to `sql<{ fn_insert_update_participant: number }[]>`